$$
\frac{1}{2}d^{{l'+1,m'-1}}\partial_xL^{{l'+1,m'-1}}\left (\vec{x} \right )
$$

In [1]:
import numpy as np
import starmap

In [2]:
sm = starmap.Starmap2D(1)

test=0.408248290463863
coefficient 0 depends on coefficient  dx  1
coefficient 0 depends on coefficient  dy  2


In [4]:
print(sm.Mx_complex)
print(sm.Mx_real)
print(sm.My_real)
print(sm.S)

[[ 0.00000000+0.j  0.40824829+0.j -0.40824829+0.j]
 [ 0.40824829+0.j  0.00000000+0.j  0.00000000+0.j]
 [-0.40824829+0.j  0.00000000+0.j  0.00000000+0.j]]
[[ 0.          0.57735027  0.        ]
 [ 0.57735027  0.          0.        ]
 [ 0.          0.          0.        ]]
[[ 0.          0.          0.57735027]
 [ 0.          0.          0.        ]
 [ 0.57735027  0.          0.        ]]
[[ 1.00000000+0.j          0.00000000+0.j          0.00000000+0.j        ]
 [ 0.00000000+0.j          0.70710678+0.j         -0.70710678+0.j        ]
 [ 0.00000000+0.j         -0.00000000-0.70710678j -0.00000000-0.70710678j]]


In [22]:
local_Mx = np.zeros((3, 3))
local_Mx[1, 0] = 0.57735027
local_Mx[0, 1] = 0.57735027
local_My = np.zeros((3, 3))
local_My[2, 0] = 0.57735027
local_My[0, 2] = 0.57735027
print(local_Mx)
print(local_My)

[[ 0.          0.57735027  0.        ]
 [ 0.57735027  0.          0.        ]
 [ 0.          0.          0.        ]]
[[ 0.          0.          0.57735027]
 [ 0.          0.          0.        ]
 [ 0.57735027  0.          0.        ]]


placements = {}
for i in range(2):
    for j in range(2):
        offset_old = sm.sga.get_offset(i, j)
        offset_new = np.array( [int(offset_old[0]*2.0), int(offset_old[1]*2.0)] )
        #print( "grid {} {} offset={} {}".format(i, j, offset_old[0], offset_old[1]) )
        print( "offset={} {}".format(offset_new[0], offset_new[1]) )
        c = sm.sga.get_grid_components(i, j)
        for unknown in 
        print(c)



----------------------------

$$
-\frac{1}{2}c^{{l'-1,m'-1}}\partial_xL^{{l'-1,m'-1}}\left (\vec{x} \right )+\frac{1}{2}d^{{l'+1,m'-1}}\partial_xL^{{l'+1,m'-1}}\left (\vec{x} \right )+\frac{1}{2}e^{{l'-1,m'+1}}\partial_xL^{{l'-1,m'+1}}\left (\vec{x} \right )-\frac{1}{2}f^{{l'+1,m'+1}}\partial_xL^{{l'+1,m'+1}}\left (\vec{x} \right )+\frac{i}{2}c^{{l'-1,m'-1}}\partial_yL^{{l'-1,m'-1}}\left (\vec{x} \right )-\frac{i}{2}d^{{l'+1,m'-1}}\partial_yL^{{l'+1,m'-1}}\left (\vec{x} \right )+\frac{i}{2}e^{{l'-1,m'+1}}\partial_yL^{{l'-1,m'+1}}\left (\vec{x} \right )-\frac{i}{2}f^{{l'+1,m'+1}}\partial_yL^{{l'+1,m'+1}}\left (\vec{x} \right )+a^{{l'-1,m'}}\partial_zL^{{l'-1,m'}}\left (\vec{x} \right )+b^{{l'+1,m'}}\partial_zL^{{l'+1,m'}}\left (\vec{x} \right )
$$

$$
\frac{1}{2}c^{{l'-1,m'-1}}\partial_xq^{{l'-1,m'-1}}\left (\vec{x} \right )
$$

$$
\frac{1}{2}0.816496580928\left(\left(-h_inv[0]\right)q^{{0,0}}\left (GridLocation2D\right )+h_inv[0]q^{{0,0}}\left (GridLocation2D\right )\right)
$$



$$
\frac{1}{2}c^{{l'-1,m'-1}}\partial_xq^{{l'-1,m'-1}}\left (\vec{x} \right )
$$
$$
-\frac{1}{2}d^{{l'+1,m'-1}}\partial_xq^{{l'+1,m'-1}}\left (\vec{x} \right )
$$
$$
-\frac{1}{2}e^{{l'-1,m'+1}}\partial_xq^{{l'-1,m'+1}}\left (\vec{x} \right )
$$
$$
\frac{1}{2}f^{{l'+1,m'+1}}\partial_xq^{{l'+1,m'+1}}\left (\vec{x} \right )
$$
$$
-\frac{i}{2}c^{{l'-1,m'-1}}\partial_yq^{{l'-1,m'-1}}\left (\vec{x} \right )
$$
$$
\frac{i}{2}d^{{l'+1,m'-1}}\partial_yq^{{l'+1,m'-1}}\left (\vec{x} \right )
$$
$$
-\frac{i}{2}e^{{l'-1,m'+1}}\partial_yq^{{l'-1,m'+1}}\left (\vec{x} \right )
$$
$$
\frac{i}{2}f^{{l'+1,m'+1}}\partial_yq^{{l'+1,m'+1}}\left (\vec{x} \right )
$$
$$
-a^{{l'-1,m'}}\partial_zq^{{l'-1,m'}}\left (\vec{x} \right )
$$
$$
-b^{{l'+1,m'}}\partial_zq^{{l'+1,m'}}\left (\vec{x} \right )
$$


$$
\frac{1}{2}c^{{l'-1,m'-1}}\partial_xq^{{l'-1,m'-1}}\left (\vec{x} \right )
$$


$$
\frac{1}{2}c^{{-1,-1}}\partial_xq^{{-1,-1}}\left (\vec{x} \right )-\frac{1}{2}d^{{1,-1}}\partial_xq^{{1,-1}}\left (\vec{x} \right )-\frac{1}{2}e^{{-1,1}}\partial_xq^{{-1,1}}\left (\vec{x} \right )+\frac{1}{2}f^{{1,1}}\partial_xq^{{1,1}}\left (\vec{x} \right )-\frac{i}{2}c^{{-1,-1}}\partial_yq^{{-1,-1}}\left (\vec{x} \right )+\frac{i}{2}d^{{1,-1}}\partial_yq^{{1,-1}}\left (\vec{x} \right )-\frac{i}{2}e^{{-1,1}}\partial_yq^{{-1,1}}\left (\vec{x} \right )+\frac{i}{2}f^{{1,1}}\partial_yq^{{1,1}}\left (\vec{x} \right )-a^{{-1,0}}\partial_zq^{{-1,0}}\left (\vec{x} \right )-b^{{1,0}}\partial_zq^{{1,0}}\left (\vec{x} \right )
$$


$$
\frac{1}{2}c^{{0,-2}}\partial_xq^{{0,-2}}\left (\vec{x} \right )-\frac{1}{2}d^{{2,-2}}\partial_xq^{{2,-2}}\left (\vec{x} \right )-\frac{1}{2}e^{{0,0}}\partial_xq^{{0,0}}\left (\vec{x} \right )+\frac{1}{2}f^{{2,0}}\partial_xq^{{2,0}}\left (\vec{x} \right )-\frac{i}{2}c^{{0,-2}}\partial_yq^{{0,-2}}\left (\vec{x} \right )+\frac{i}{2}d^{{2,-2}}\partial_yq^{{2,-2}}\left (\vec{x} \right )-\frac{i}{2}e^{{0,0}}\partial_yq^{{0,0}}\left (\vec{x} \right )+\frac{i}{2}f^{{2,0}}\partial_yq^{{2,0}}\left (\vec{x} \right )-a^{{0,-1}}\partial_zq^{{0,-1}}\left (\vec{x} \right )-b^{{2,-1}}\partial_zq^{{2,-1}}\left (\vec{x} \right )
$$


$$
\frac{1}{2}c^{{0,0}}\partial_xq^{{0,0}}\left (\vec{x} \right )-\frac{1}{2}d^{{2,0}}\partial_xq^{{2,0}}\left (\vec{x} \right )-\frac{1}{2}e^{{0,2}}\partial_xq^{{0,2}}\left (\vec{x} \right )+\frac{1}{2}f^{{2,2}}\partial_xq^{{2,2}}\left (\vec{x} \right )-\frac{i}{2}c^{{0,0}}\partial_yq^{{0,0}}\left (\vec{x} \right )+\frac{i}{2}d^{{2,0}}\partial_yq^{{2,0}}\left (\vec{x} \right )-\frac{i}{2}e^{{0,2}}\partial_yq^{{0,2}}\left (\vec{x} \right )+\frac{i}{2}f^{{2,2}}\partial_yq^{{2,2}}\left (\vec{x} \right )-a^{{0,1}}\partial_zq^{{0,1}}\left (\vec{x} \right )-b^{{2,1}}\partial_zq^{{2,1}}\left (\vec{x} \right )
$$



-----------------
$$
0-0.408248290464\partial_xq^{{1,-1}}\left (\vec{x} \right )
$$
$$
-0.408248290464\left(\left(-h_inv[0]\right)q^{{1,-1}}\left (GridLocation2D\right )+h_inv[0]q^{{1,-1}}\left (GridLocation2D\right )\right)
$$

-----------------
$$
0-0.4472135955\partial_xq^{{2,-2}}\left (\vec{x} \right )
$$

-----------------
$$
0.408248290464\partial_xq^{{0,0}}\left (\vec{x} \right )-0.182574185835\partial_xq^{{2,0}}\left (\vec{x} \right )
$$
$$
0.408248290464\left(\left(-h_inv[0]\right)q^{{0,0}}\left (GridLocation2D\right )+h_inv[0]q^{{0,0}}\left (GridLocation2D\right )\right)
$$
